src/uct/cuda/base/cuda_md.c

```diff
@@ -64,10 +65,16 @@ void uct_cuda_base_get_sys_dev(CUdevice cuda_device,
 ucs_status_t
 uct_cuda_base_get_cuda_device(ucs_sys_device_t sys_dev, CUdevice *device)
 {
-    uintptr_t value = ucs_topo_sysdev_get_user_value(sys_dev);
+    uintptr_t value;
+    ucs_status_t status;
+
+    status = ucs_topo_sys_dev_get_user_value(sys_dev, &value);
+    if (status != UCS_OK) {
+        return UCS_ERR_NO_DEVICE;
+    }
 
     *device = value;
-    if ((value == UINTPTR_MAX) || (*device == CU_DEVICE_INVALID)) {
+    if (*device == CU_DEVICE_INVALID) {
```

Comment:
IMO we should check value for UINTPTR_MAX before doing *device=value
maybe rename value to user_value